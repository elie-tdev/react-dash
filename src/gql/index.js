import { gql } from '@apollo/client';

export const queries = {
  InitializeUser: gql`
    query InitializeUser($contact_id: String!) {
      Contacts_Get(contact_id: $contact_id) {
        payload {
          data {
            Firstname1
            Lastname1
          }
        }
      }
      Contacts_GetLoans(contact_id: $contact_id) {
        payload {
          data {
            Acctrefno
            Loan_Number
            Current_Payoff_Balance
            Total_Current_Due_Balance
            Loan_Type
          }
        }
      }
    }
  `,
  UpdateBankAccounts: gql`
    mutation UpdateBankAccounts($payload: String!) {
      NlsCore_ImportXml(payload: $payload) {
        payload {
          data
        }
      }
    }
  `,
  BankAccounts: gql`
    query BankAccountsGet($contact_id: String!) {
      Contacts_GetFinancials(contact_id: $contact_id) {
        payload {
          data {
            BankAccounts
          }
        }
      }
      Contacts_Get(contact_id: $contact_id) {
        payload {
          data {
            Cifnumber
          }
        }
      }
    }
  `,
  UpdateAddress: gql`
    mutation UpdateAddress($payload: String!) {
      NlsCore_ImportXml(payload: $payload) {
        payload {
          data
        }
      }
    }
  `,
  ChangeAddress: gql`
    query ChangeAddressGet($contact_id: String!) {
      Contacts_Get(contact_id: $contact_id) {
        payload {
          data {
            Cifnumber
            Cifno
            Street_Address1
            Street_Address2
            City
            State
            Zip
            Entity
          }
        }
      }
    }
  `,
  SubmitOneTimePayment: gql`
    mutation SubmitOneTimePayment($payload: String!) {
      NlsCore_ImportXml(payload: $payload) {
        payload {
          data
        }
      }
    }
  `,
  CancelAutopayPayment: gql`
    mutation CancelAutopayPayment($payload: String!) {
      NlsCore_ImportXml(payload: $payload) {
        payload {
          data
        }
      }
    }
  `,
  LoanOverview: gql`
    query LoanGet($loan_id: String!) {
      Loans_Get(loan_id: $loan_id) {
        payload {
          data {
            Acctrefno
            Loan_Number
            Current_Payoff_Balance
            Original_Note_Amount
            Total_Current_Due_Balance
            Total_Past_Due_Balance
            Days_Past_Due
            Current_Principal_Balance
            Current_Interest_Balance
            Current_Interest_Rate
            Current_Late_Charge_Balance
            Current_Fees_Balance
          }
        }
      }
      Loans_GetPaymentInfo(loan_id: $loan_id) {
        payload {
          data {
            Amortized_Payment_Amount
            Current_Principal_Payment_Date
            Next_Principal_Payment_Date
          }
        }
      }
    }
  `,
  Balance: gql`
    query LoanGet($loan_id: String!) {
      Loans_Get(loan_id: $loan_id) {
        payload {
          data {
            Acctrefno
            Loan_Number
            Current_Payoff_Balance
            Original_Note_Amount
            Total_Current_Due_Balance
            Total_Past_Due_Balance
            Current_Principal_Balance
            Current_Interest_Balance
            Current_Interest_Rate
            Current_Fees_Balance
            Current_Late_Charge_Balance
            Open_Date
            Open_Maturity_Date
            Days_Past_Due
          }
        }
      }
      Loans_GetPaymentInfo(loan_id: $loan_id) {
        payload {
          data {
            Total_Payments
            Amortized_Payment_Amount
            Last_Payment_Date
            Last_Payment_Amount
            Current_Principal_Payment_Date
            Next_Principal_Payment_Date
          }
        }
      }
      Loans_GetSetup(loan_id: $loan_id) {
        payload {
          data {
            Principal_Period
          }
        }
      }
    }
  `,
  payoffDetails: gql`
    query LoanGetPayoffDetails($loan_id: String!, $payoffDate: String!) {
      Loans_GetPayoffDetails(loan_id: $loan_id, payoffDate: $payoffDate) {
        payload {
          data {
            Payoount
          }
        }
      }
    }
  `,
  PaymentHistory: gql`
    query LoanGet($loan_id: String!) {
      Loans_GetPaymentInfo(loan_id: $loan_id) {
        payload {
          data {
            Amortized_Payment_Amount
            Current_Principal_Payment_Date
            Next_Principal_Payment_Date
          }
        }
      }
      Loans_GetPaymentHistory(loan_id: $loan_id) {
        payload {
          data {
            Row_Id
            Payment_Reference_No
            Date_Paid
            Payment_Amount
            Payment_Type
            Payment_Description
            Nsf_Flag
            Transaction_Reference_No
            Account_Number
          }
        }
      }
    }
  `,
  useLoanState: gql`
    query LoanGet($loan_id: String!) {
      Loans_GetAutomatedPayments(loan_id: $loan_id) {
        payload {
          data {
            Row_Id
            Payment_Type
            Status
            Billing_Type
          }
        }
      }
      Loans_GetPaymentHistory(loan_id: $loan_id) {
        payload {
          data {
            Payment_Type
            Date_Paid
          }
        }
      }
      Loans_GetPaymentInfo(loan_id: $loan_id) {
        payload {
          data {
            Amortized_Payment_Amount
            Current_Principal_Payment_Date
            Next_Principal_Payment_Date
          }
        }
      }
      Loans_GetSetup(loan_id: $loan_id) {
        payload {
          data {
            Principal_Period
          }
        }
      }
      Loans_GetPaymentsDue(loan_id: $loan_id) {
        payload {
          data {
            Row_Id
            Transaction_Code
            Date_Due
            Payment_Amount
          }
        }
      }
      Loans_Get(loan_id: $loan_id) {
        payload {
          data {
            Days_Past_Due
            Current_Payoff_Balance
            Total_Current_Due_Balance
            Total_Past_Due_Balance
            Original_Note_Amount
            Loan_Number
          }
        }
      }
      Loans_GetDetail2(loan_id: $loan_id) {
        payload {
          data {
            ACHCompanyID
          }
        }
      }
    }
  `,
  GetAutomatedPaymentsByRowId: gql`
    query getAutomatedPaymentsByRowId($row_id: String!) {
      AutomatedPayments_Get(row_id: $row_id) {
        payload {
          data {
            Row_Id
            Billing_Period
            Amount_Fixed
            Billing_Next_Date
            Account_Number
            Account_Type
            Payment_Type
            Status
            Billing_Type
            Billing_Start_Date
            Billing_Next_Date
            Aba_Number
            AutomatedPaymentCompanyID
          }
        }
      }
    }
  `,
  CurrentState: gql`
    query getCurrentUser($loan_id: String!) {
      Loans_Get(loan_id: $loan_id) {
        payload {
          data {
            Days_Past_Due
          }
        }
      }
    }
  `,
  Statements: gql`
    query getStatements($loan_id: String!) {
      Loans_GetStatements(loan_id: $loan_id) {
        payload {
          data {
            Statement_Row_Id
            Statement_Date
            Statement_Due_Date
            Current_Rate
            Total_Past_Due_Balance
            Total_Current_Due_Balance
            Current_Payoff_Balance
          }
        }
      }
    }
  `,
  StatementFile: gql`
    query getStatementFile($statement_id: String!) {
      LoanStatements_GetFile(statement_id: $statement_id) {
        payload {
          data {
            File
          }
        }
      }
    }
  `,
  ContactUs: gql`
    query LoanGet($loan_id: String!) {
      Loans_Get(loan_id: $loan_id) {
        payload {
          data {
            Portfolio_Code_Id
          }
        }
      }
    }
  `,
};
