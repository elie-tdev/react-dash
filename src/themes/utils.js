export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
