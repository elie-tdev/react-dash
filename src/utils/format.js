const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
});

const parseFloat = (str, val = 2) => {
  str = str && str.toString();
  str = str && str.indexOf('.') > -1 ? str.slice(0, str.indexOf('.') + val + 1) : str;
  return Number(str);
};

export const formatDollar = number => formatter.format(parseFloat(number, 2));
