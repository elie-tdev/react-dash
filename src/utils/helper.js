const maskAccountNumber = accNumber => {
  return accNumber && <>&bull;&bull;&bull;&bull; {accNumber.slice(accNumber.length - 4)}</>;
};

export default maskAccountNumber;
