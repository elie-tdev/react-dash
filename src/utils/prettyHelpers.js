export const prettyCurrency = (amount, currency = 'USD') => {
  if (amount == null) {
    return false;
  }
  if (amount < 0) {
    amount = 0;
  }
  return amount.toLocaleString('en', { style: 'currency', currency });
};

export const prettyDateTime = time => {
  if (time == null) {
    return false;
  }
  let d = new Date(time);
  return (
    d.toLocaleDateString('en', {
      month: '2-digit',
      day: '2-digit',
      year: '2-digit',
    }) +
    ' at ' +
    d.toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' }).replace(/^0(?:0:0?)?/, '')
  );
};

export const prettyDate = time => new Date(time).toLocaleDateString('en', {});

export const prettyTime = time =>
  new Date(time).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });

export const prettyHour = time =>
  new Date(time)
    .toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' })
    .replace(/^0(?:0:0?)?/, '');

export const prettyPhone = number =>
  number.substr(2).replace(/(\d{1,2})(\d{1})?(\d{1,3})?(\d{1,4})?/, function (_, p1, p2, p3, p4) {
    let output = '';
    if (p1) output = `${p1}`;
    if (p2) output += `${p2}-`;
    if (p3) output += `${p3}-`;
    if (p4) output += `${p4}`;
    return output;
  });

export const minimumDays = (date, numDays) => {
  const d = new Date(date);
  const totalNumDays = daysIntoYear(d) + numDays;
  return Boolean(daysIntoYear(d) >= totalNumDays || totalNumDays <= 365);
};

export const daysIntoYear = date => {
  return (
    (Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()) -
      Date.UTC(date.getFullYear(), 0, 0)) /
    24 /
    60 /
    60 /
    1000
  );
};

export const cleanAmountRegex = value => {
  return value.replace(/[^0-9]/g, '');
};

export const cleanAmount = value => {
  const pos = value.indexOf('.');
  let result;
  if (pos !== -1) {
    const part1 = value.substr(0, pos);
    const part2 = value.substr(pos + 1);
    result = cleanAmountRegex(part1) + '.' + cleanAmountRegex(part2);
  } else {
    result = cleanAmountRegex(value);
  }
  return result;
};
