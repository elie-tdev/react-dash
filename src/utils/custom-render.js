import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
