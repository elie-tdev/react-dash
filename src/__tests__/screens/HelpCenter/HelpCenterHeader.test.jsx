import React from 'react';

import HelpCenterHeader from '@/screens/HelpCenter/HelpCenterHeader';
import { render } from '@/utils/custom-render';
import { queries } from '@/gql';

const getLoanFplus = {
  request: {
    query: queries.ContactUs,
    variables: {
      loan_id: '337720',
    },
  },
  result: {
    data: {
      Loans_Get: {
        payload: {
          data: {
            Portfolio_Code_Id: 0,
          },
        },
      },
    },
  },
};

describe('HelpCenterHeader', () => {
  test('renders HelpCenterHeader', async () => {
    const { findByText, container } = render(<HelpCenterHeader />, {
      mocks: [getLoanFplus],
    });
    expect(await findByText('Contact Us')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
