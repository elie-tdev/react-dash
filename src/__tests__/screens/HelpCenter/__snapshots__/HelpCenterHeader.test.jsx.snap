// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpCenterHeader renders HelpCenterHeader 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiBox-root MuiBox-root-34 makeStyles-row-12 makeStyles-header-1"
    >
      <div
        class="MuiBox-root MuiBox-root-35"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root makeStyles-arrowBackIconButton-3"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-backIcon-2"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h4
          class="MuiTypography-root makeStyles-pageTitle-4 MuiTypography-h4"
        >
          Contact Us
        </h4>
      </div>
    </div>
  </div>
</div>
`;
